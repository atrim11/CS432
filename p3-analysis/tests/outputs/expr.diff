--- outputs/expr.txt	2024-10-23 16:59:50.006625415 -0400
+++ expected/expr.txt	2024-10-23 14:21:17.462651764 -0400
@@ -1,193 +1,221 @@
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-Invalid: Variable 'result' declared as void on line 2
-Error: Variable 'a' used without being defined on line 5
-Error: Variable 'b' used without being defined on line 5
-Error: Variable 'c' used without being defined on line 5
-Error: Variable 'd' used without being defined on line 5
-Error: Variable 'e' used without being defined on line 5
-Error: Variable 'f' used without being defined on line 5
-Error: Variable 'g' used without being defined on line 5
-Error: Variable 'h' used without being defined on line 5
-Error: Variable 'i' used without being defined on line 5
-Error: Variable 'j' used without being defined on line 5
-Error: Variable 'k' used without being defined on line 5
-Error: Variable 'l' used without being defined on line 5
-Error: Variable 'm' used without being defined on line 5
-Error: Variable 'n' used without being defined on line 5
-Error: Variable 'o' used without being defined on line 5
-Error: Variable 'p' used without being defined on line 5
-Error: Variable 'q' used without being defined on line 5
-Error: Variable 'r' used without being defined on line 5
-Error: Variable 's' used without being defined on line 5
-Error: Variable 't' used without being defined on line 5
-Error: Variable 'u' used without being defined on line 5
-Error: Variable 'v' used without being defined on line 5
-Error: Variable 'w' used without being defined on line 5
-Error: Variable 'x' used without being defined on line 5
-Error: Variable 'y' used without being defined on line 5
+Void variable 'result' on line 2
+Symbol 'a' undefined on line 5
+Symbol 'b' undefined on line 5
+Symbol 'c' undefined on line 5
+Symbol 'd' undefined on line 5
+Symbol 'e' undefined on line 5
+Symbol 'f' undefined on line 5
+Type mismatch: bool expected but int found on line 5
+Symbol 'g' undefined on line 5
+Symbol 'h' undefined on line 5
+Symbol 'i' undefined on line 5
+Symbol 'j' undefined on line 5
+Symbol 'k' undefined on line 5
+Symbol 'l' undefined on line 5
+Symbol 'm' undefined on line 5
+Symbol 'n' undefined on line 5
+Symbol 'o' undefined on line 5
+Symbol 'p' undefined on line 5
+Symbol 'q' undefined on line 5
+Symbol 'r' undefined on line 5
+Symbol 's' undefined on line 5
+Symbol 't' undefined on line 5
+Symbol 'u' undefined on line 5
+Type mismatch: int expected but bool found on line 5
+Type mismatch: int expected but bool found on line 5
+Symbol 'v' undefined on line 5
+Symbol 'w' undefined on line 5
+Symbol 'x' undefined on line 5
+Symbol 'y' undefined on line 5
+Type mismatch: bool expected but int found on line 5
 Symbol 'foo' undefined on line 6
 Symbol 'bar' undefined on line 6
-Error: Variable 'arr' used without being defined on line 6
-Error: Variable 'arr' used without being defined on line 6
+Symbol 'arr' undefined on line 6
+Symbol 'arr' undefined on line 6
 Symbol 'baz' undefined on line 6
-Error: Variable 'x' used without being defined on line 6
-Error: Variable 'y' used without being defined on line 6
-Error: Variable 'z' used without being defined on line 6
-Error: Variable 'a' used without being defined on line 7
-Error: Variable 'b' used without being defined on line 7
-Error: Variable 'c' used without being defined on line 7
-Error: Variable 'd' used without being defined on line 7
-Error: Variable 'e' used without being defined on line 7
-Error: Variable 'f' used without being defined on line 7
-Error: Variable 'g' used without being defined on line 7
-Error: Variable 'h' used without being defined on line 7
-Error: Variable 'i' used without being defined on line 7
-Error: Variable 'j' used without being defined on line 7
+Type mismatch: bool expected but int found on line 6
+Symbol 'x' undefined on line 6
+Symbol 'y' undefined on line 6
+Symbol 'z' undefined on line 6
+Type mismatch: int expected but bool found on line 5
+Type mismatch: int expected but bool found on line 6
+Symbol 'a' undefined on line 7
+Symbol 'b' undefined on line 7
+Type mismatch: bool expected but int found on line 7
+Symbol 'c' undefined on line 7
+Symbol 'd' undefined on line 7
+Symbol 'e' undefined on line 7
+Symbol 'f' undefined on line 7
+Type mismatch: int expected but bool found on line 7
+Symbol 'g' undefined on line 7
+Symbol 'h' undefined on line 7
+Symbol 'i' undefined on line 7
+Symbol 'j' undefined on line 7
+Type mismatch: int expected but bool found on line 7
 Symbol 'alpha' undefined on line 8
 Symbol 'beta' undefined on line 8
 Symbol 'gamma' undefined on line 8
-Error: Variable 'delta' used without being defined on line 8
-Error: Variable 'x' used without being defined on line 8
-Error: Variable 'm' used without being defined on line 8
-Error: Variable 'n' used without being defined on line 8
-Error: Variable 'p' used without being defined on line 8
-Error: Variable 'q' used without being defined on line 8
+Symbol 'delta' undefined on line 8
+Symbol 'x' undefined on line 8
+Type mismatch: int expected but bool found on line 8
+Symbol 'm' undefined on line 8
+Symbol 'n' undefined on line 8
+Symbol 'p' undefined on line 8
+Symbol 'q' undefined on line 8
+Type mismatch: int expected but bool found on line 8
 Symbol 'foo' undefined on line 9
-Error: Variable 'bar' used without being defined on line 9
-Error: Variable 'a' used without being defined on line 9
-Error: Variable 'b' used without being defined on line 9
+Symbol 'bar' undefined on line 9
+Type mismatch: int expected but bool found on line 9
+Type mismatch: int expected but bool found on line 9
+Symbol 'a' undefined on line 9
+Symbol 'b' undefined on line 9
 Symbol 'funcA' undefined on line 10
 Symbol 'funcB' undefined on line 10
-Error: Variable 'x' used without being defined on line 10
-Error: Variable 'y' used without being defined on line 10
-Error: Variable 'z' used without being defined on line 10
-Error: Variable 'array' used without being defined on line 10
-Error: Variable 'array' used without being defined on line 10
-Error: Variable 'array' used without being defined on line 10
-Error: Variable 'x' used without being defined on line 11
-Error: Variable 'y' used without being defined on line 11
-Error: Variable 'z' used without being defined on line 11
+Symbol 'x' undefined on line 10
+Symbol 'y' undefined on line 10
+Symbol 'z' undefined on line 10
+Type mismatch: bool expected but int found on line 10
+Symbol 'array' undefined on line 10
+Symbol 'array' undefined on line 10
+Symbol 'array' undefined on line 10
+Type mismatch: int expected but bool found on line 9
+Type mismatch: int expected but bool found on line 10
+Symbol 'x' undefined on line 11
+Symbol 'y' undefined on line 11
+Symbol 'z' undefined on line 11
 Symbol 'funcC' undefined on line 11
-Error: Variable 'a' used without being defined on line 11
-Error: Variable 'b' used without being defined on line 11
-Error: Variable 'c' used without being defined on line 11
-Error: Variable 'd' used without being defined on line 11
-Error: Variable 'foo' used without being defined on line 11
-Error: Variable 'bar' used without being defined on line 11
-Error: Variable 'arr' used without being defined on line 12
-Error: Variable 'arr' used without being defined on line 12
+Symbol 'a' undefined on line 11
+Symbol 'b' undefined on line 11
+Symbol 'c' undefined on line 11
+Symbol 'd' undefined on line 11
+Type mismatch: bool expected but int found on line 11
+Symbol 'foo' undefined on line 11
+Symbol 'bar' undefined on line 11
+Type mismatch: int expected but bool found on line 9
+Type mismatch: int expected but bool found on line 11
+Symbol 'arr' undefined on line 12
+Symbol 'arr' undefined on line 12
+Type mismatch: int expected but bool found on line 12
 Symbol 'foo' undefined on line 12
 Symbol 'bar' undefined on line 12
-Error: Variable 'x' used without being defined on line 13
-Error: Variable 'y' used without being defined on line 13
-Error: Variable 'z' used without being defined on line 13
-Error: Variable 'arr' used without being defined on line 13
+Type mismatch: int is incompatible with bool on line 12
+Symbol 'x' undefined on line 13
+Symbol 'y' undefined on line 13
+Symbol 'z' undefined on line 13
+Symbol 'arr' undefined on line 13
 Symbol 'funcD' undefined on line 13
+Type mismatch: int expected but bool found on line 12
+Type mismatch: int expected but bool found on line 13
 Symbol 'foo' undefined on line 14
-Error: Variable 'a' used without being defined on line 14
+Symbol 'a' undefined on line 14
 Symbol 'bar' undefined on line 14
-Error: Variable 'b' used without being defined on line 14
-Error: Variable 'c' used without being defined on line 14
+Symbol 'b' undefined on line 14
+Symbol 'c' undefined on line 14
 Symbol 'baz' undefined on line 14
-Error: Variable 'd' used without being defined on line 14
-Error: Variable 'x' used without being defined on line 14
-Error: Variable 'y' used without being defined on line 14
-Error: Variable 'e' used without being defined on line 14
-Error: Variable 'f' used without being defined on line 14
+Symbol 'd' undefined on line 14
+Symbol 'x' undefined on line 14
+Symbol 'y' undefined on line 14
+Symbol 'e' undefined on line 14
+Symbol 'f' undefined on line 14
+Type mismatch: int expected but bool found on line 14
+Type mismatch: bool expected but int found on line 15
+Type mismatch: bool expected but int found on line 15
+Type mismatch: int is incompatible with bool on line 15
+Type mismatch: int is incompatible with bool on line 15
+Type mismatch: int expected but bool found on line 15
+Type mismatch: int expected but bool found on line 9
+Type mismatch: int is incompatible with bool on line 9
 Symbol 'foo' undefined on line 15
-Error: Variable 'arr' used without being defined on line 15
+Symbol 'arr' undefined on line 15
 Symbol 'bar' undefined on line 15
-Error: Variable 'arr' used without being defined on line 15
-Error: Variable 'alpha' used without being defined on line 16
-Error: Variable 'beta' used without being defined on line 16
-Error: Variable 'gamma' used without being defined on line 16
-Error: Variable 'delta' used without being defined on line 16
-Error: Variable 'epsilon' used without being defined on line 16
-Error: Variable 'z' used without being defined on line 16
-Error: Variable 'a' used without being defined on line 16
-Error: Variable 'b' used without being defined on line 16
-Error: Variable 'c' used without being defined on line 16
+Symbol 'arr' undefined on line 15
+Symbol 'alpha' undefined on line 16
+Symbol 'beta' undefined on line 16
+Symbol 'gamma' undefined on line 16
+Symbol 'delta' undefined on line 16
+Symbol 'epsilon' undefined on line 16
+Symbol 'z' undefined on line 16
+Symbol 'a' undefined on line 16
+Type mismatch: bool expected but int found on line 16
+Symbol 'b' undefined on line 16
+Symbol 'c' undefined on line 16
+Type mismatch: int expected but bool found on line 15
+Type mismatch: int expected but bool found on line 16
 Symbol 'funcA' undefined on line 17
-Error: Variable 'x' used without being defined on line 17
-Error: Variable 'arr' used without being defined on line 17
-Error: Variable 'arr' used without being defined on line 17
+Symbol 'x' undefined on line 17
+Symbol 'arr' undefined on line 17
+Symbol 'arr' undefined on line 17
 Symbol 'funcB' undefined on line 17
-Error: Variable 'y' used without being defined on line 17
-Error: Variable 'x' used without being defined on line 18
-Error: Variable 'y' used without being defined on line 18
+Symbol 'y' undefined on line 17
+Type mismatch: int expected but bool found on line 17
+Symbol 'x' undefined on line 18
+Symbol 'y' undefined on line 18
 Symbol 'foo' undefined on line 18
-Error: Variable 'bar' used without being defined on line 18
-Error: Variable 'a' used without being defined on line 19
-Error: Variable 'b' used without being defined on line 19
-Error: Variable 'c' used without being defined on line 19
-Error: Variable 'd' used without being defined on line 19
+Symbol 'bar' undefined on line 18
+Type mismatch: bool expected but int found on line 18
+Type mismatch: int expected but bool found on line 18
+Symbol 'a' undefined on line 19
+Symbol 'b' undefined on line 19
+Symbol 'c' undefined on line 19
+Symbol 'd' undefined on line 19
 Symbol 'foo' undefined on line 19
 Symbol 'bar' undefined on line 19
+Type mismatch: int expected but bool found on line 18
+Type mismatch: int expected but bool found on line 19
 Symbol 'foo' undefined on line 20
-Error: Variable 'a' used without being defined on line 20
-Error: Variable 'b' used without being defined on line 20
-Error: Variable 'c' used without being defined on line 20
-Error: Variable 'a' used without being defined on line 21
-Error: Variable 'b' used without being defined on line 21
+Symbol 'a' undefined on line 20
+Symbol 'b' undefined on line 20
+Symbol 'c' undefined on line 20
+Type mismatch: bool expected but int found on line 20
+Type mismatch: bool expected but int found on line 20
+Type mismatch: bool expected but int found on line 20
+Type mismatch: int expected but bool found on line 20
+Type mismatch: int expected but bool found on line 20
+Symbol 'a' undefined on line 21
+Symbol 'b' undefined on line 21
 Symbol 'foo' undefined on line 21
 Symbol 'bar' undefined on line 21
-Error: Variable 'x' used without being defined on line 21
-Error: Variable 'y' used without being defined on line 21
+Type mismatch: int expected but bool found on line 21
+Symbol 'x' undefined on line 21
+Symbol 'y' undefined on line 21
+Type mismatch: int expected but bool found on line 21
+Type mismatch: int expected but bool found on line 9
 Symbol 'foo' undefined on line 22
-Error: Variable 'a' used without being defined on line 22
+Symbol 'a' undefined on line 22
 Symbol 'bar' undefined on line 22
-Error: Variable 'b' used without being defined on line 22
-Error: Variable 'arr' used without being defined on line 23
-Error: Variable 'arr' used without being defined on line 23
-Error: Variable 'arr' used without being defined on line 23
-Error: Variable 'arr' used without being defined on line 23
+Symbol 'b' undefined on line 22
+Type mismatch: int is incompatible with bool on line 22
+Symbol 'arr' undefined on line 23
+Symbol 'arr' undefined on line 23
+Symbol 'arr' undefined on line 23
+Symbol 'arr' undefined on line 23
+Type mismatch: int expected but bool found on line 23
 Symbol 'funcC' undefined on line 23
-Error: Variable 'x' used without being defined on line 23
+Symbol 'x' undefined on line 23
 Symbol 'funcD' undefined on line 23
-Error: Variable 'y' used without being defined on line 23
+Symbol 'y' undefined on line 23
+Type mismatch: bool expected but int found on line 24
+Type mismatch: bool expected but int found on line 24
+Type mismatch: int expected but bool found on line 24
 Symbol 'foo' undefined on line 24
 Symbol 'bar' undefined on line 24
-Error: Variable 'x' used without being defined on line 25
-Error: Variable 'y' used without being defined on line 25
-Error: Variable 'z' used without being defined on line 25
+Type mismatch: int is incompatible with bool on line 24
+Type mismatch: int expected but bool found on line 23
+Type mismatch: int expected but bool found on line 24
+Symbol 'x' undefined on line 25
+Symbol 'y' undefined on line 25
+Symbol 'z' undefined on line 25
 Symbol 'funcC' undefined on line 25
-Error: Variable 'a' used without being defined on line 25
-Error: Variable 'b' used without being defined on line 25
-Error: Variable 'c' used without being defined on line 25
-Error: Variable 'd' used without being defined on line 25
-Error: Variable 'foo' used without being defined on line 25
-Error: Variable 'bar' used without being defined on line 25
-Type mismatch in assignment on line 4: expected void, got ???
+Symbol 'a' undefined on line 25
+Symbol 'b' undefined on line 25
+Symbol 'c' undefined on line 25
+Symbol 'd' undefined on line 25
+Type mismatch: bool expected but int found on line 25
+Symbol 'foo' undefined on line 25
+Symbol 'bar' undefined on line 25
+Type mismatch: int expected but bool found on line 25
+Type mismatch: bool expected but int found on line 23
+Type mismatch: bool expected but int found on line 23
+Type mismatch: int expected but bool found on line 22
+Type mismatch: int expected but bool found on line 23
+Type mismatch: void is incompatible with bool on line 4
