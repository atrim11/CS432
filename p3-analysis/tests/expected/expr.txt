Void variable 'result' on line 2
Symbol 'a' undefined on line 5
Symbol 'b' undefined on line 5
Symbol 'c' undefined on line 5
Symbol 'd' undefined on line 5
Symbol 'e' undefined on line 5
Symbol 'f' undefined on line 5
Type mismatch: bool expected but int found on line 5
Symbol 'g' undefined on line 5
Symbol 'h' undefined on line 5
Symbol 'i' undefined on line 5
Symbol 'j' undefined on line 5
Symbol 'k' undefined on line 5
Symbol 'l' undefined on line 5
Symbol 'm' undefined on line 5
Symbol 'n' undefined on line 5
Symbol 'o' undefined on line 5
Symbol 'p' undefined on line 5
Symbol 'q' undefined on line 5
Symbol 'r' undefined on line 5
Symbol 's' undefined on line 5
Symbol 't' undefined on line 5
Symbol 'u' undefined on line 5
Type mismatch: int expected but bool found on line 5
Type mismatch: int expected but bool found on line 5
Symbol 'v' undefined on line 5
Symbol 'w' undefined on line 5
Symbol 'x' undefined on line 5
Symbol 'y' undefined on line 5
Type mismatch: bool expected but int found on line 5
Symbol 'foo' undefined on line 6
Symbol 'bar' undefined on line 6
Symbol 'arr' undefined on line 6
Symbol 'arr' undefined on line 6
Symbol 'baz' undefined on line 6
Type mismatch: bool expected but int found on line 6
Symbol 'x' undefined on line 6
Symbol 'y' undefined on line 6
Symbol 'z' undefined on line 6
Type mismatch: int expected but bool found on line 5
Type mismatch: int expected but bool found on line 6
Symbol 'a' undefined on line 7
Symbol 'b' undefined on line 7
Type mismatch: bool expected but int found on line 7
Symbol 'c' undefined on line 7
Symbol 'd' undefined on line 7
Symbol 'e' undefined on line 7
Symbol 'f' undefined on line 7
Type mismatch: int expected but bool found on line 7
Symbol 'g' undefined on line 7
Symbol 'h' undefined on line 7
Symbol 'i' undefined on line 7
Symbol 'j' undefined on line 7
Type mismatch: int expected but bool found on line 7
Symbol 'alpha' undefined on line 8
Symbol 'beta' undefined on line 8
Symbol 'gamma' undefined on line 8
Symbol 'delta' undefined on line 8
Symbol 'x' undefined on line 8
Type mismatch: int expected but bool found on line 8
Symbol 'm' undefined on line 8
Symbol 'n' undefined on line 8
Symbol 'p' undefined on line 8
Symbol 'q' undefined on line 8
Type mismatch: int expected but bool found on line 8
Symbol 'foo' undefined on line 9
Symbol 'bar' undefined on line 9
Type mismatch: int expected but bool found on line 9
Type mismatch: int expected but bool found on line 9
Symbol 'a' undefined on line 9
Symbol 'b' undefined on line 9
Symbol 'funcA' undefined on line 10
Symbol 'funcB' undefined on line 10
Symbol 'x' undefined on line 10
Symbol 'y' undefined on line 10
Symbol 'z' undefined on line 10
Type mismatch: bool expected but int found on line 10
Symbol 'array' undefined on line 10
Symbol 'array' undefined on line 10
Symbol 'array' undefined on line 10
Type mismatch: int expected but bool found on line 9
Type mismatch: int expected but bool found on line 10
Symbol 'x' undefined on line 11
Symbol 'y' undefined on line 11
Symbol 'z' undefined on line 11
Symbol 'funcC' undefined on line 11
Symbol 'a' undefined on line 11
Symbol 'b' undefined on line 11
Symbol 'c' undefined on line 11
Symbol 'd' undefined on line 11
Type mismatch: bool expected but int found on line 11
Symbol 'foo' undefined on line 11
Symbol 'bar' undefined on line 11
Type mismatch: int expected but bool found on line 9
Type mismatch: int expected but bool found on line 11
Symbol 'arr' undefined on line 12
Symbol 'arr' undefined on line 12
Type mismatch: int expected but bool found on line 12
Symbol 'foo' undefined on line 12
Symbol 'bar' undefined on line 12
Type mismatch: int is incompatible with bool on line 12
Symbol 'x' undefined on line 13
Symbol 'y' undefined on line 13
Symbol 'z' undefined on line 13
Symbol 'arr' undefined on line 13
Symbol 'funcD' undefined on line 13
Type mismatch: int expected but bool found on line 12
Type mismatch: int expected but bool found on line 13
Symbol 'foo' undefined on line 14
Symbol 'a' undefined on line 14
Symbol 'bar' undefined on line 14
Symbol 'b' undefined on line 14
Symbol 'c' undefined on line 14
Symbol 'baz' undefined on line 14
Symbol 'd' undefined on line 14
Symbol 'x' undefined on line 14
Symbol 'y' undefined on line 14
Symbol 'e' undefined on line 14
Symbol 'f' undefined on line 14
Type mismatch: int expected but bool found on line 14
Type mismatch: bool expected but int found on line 15
Type mismatch: bool expected but int found on line 15
Type mismatch: int is incompatible with bool on line 15
Type mismatch: int is incompatible with bool on line 15
Type mismatch: int expected but bool found on line 15
Type mismatch: int expected but bool found on line 9
Type mismatch: int is incompatible with bool on line 9
Symbol 'foo' undefined on line 15
Symbol 'arr' undefined on line 15
Symbol 'bar' undefined on line 15
Symbol 'arr' undefined on line 15
Symbol 'alpha' undefined on line 16
Symbol 'beta' undefined on line 16
Symbol 'gamma' undefined on line 16
Symbol 'delta' undefined on line 16
Symbol 'epsilon' undefined on line 16
Symbol 'z' undefined on line 16
Symbol 'a' undefined on line 16
Type mismatch: bool expected but int found on line 16
Symbol 'b' undefined on line 16
Symbol 'c' undefined on line 16
Type mismatch: int expected but bool found on line 15
Type mismatch: int expected but bool found on line 16
Symbol 'funcA' undefined on line 17
Symbol 'x' undefined on line 17
Symbol 'arr' undefined on line 17
Symbol 'arr' undefined on line 17
Symbol 'funcB' undefined on line 17
Symbol 'y' undefined on line 17
Type mismatch: int expected but bool found on line 17
Symbol 'x' undefined on line 18
Symbol 'y' undefined on line 18
Symbol 'foo' undefined on line 18
Symbol 'bar' undefined on line 18
Type mismatch: bool expected but int found on line 18
Type mismatch: int expected but bool found on line 18
Symbol 'a' undefined on line 19
Symbol 'b' undefined on line 19
Symbol 'c' undefined on line 19
Symbol 'd' undefined on line 19
Symbol 'foo' undefined on line 19
Symbol 'bar' undefined on line 19
Type mismatch: int expected but bool found on line 18
Type mismatch: int expected but bool found on line 19
Symbol 'foo' undefined on line 20
Symbol 'a' undefined on line 20
Symbol 'b' undefined on line 20
Symbol 'c' undefined on line 20
Type mismatch: bool expected but int found on line 20
Type mismatch: bool expected but int found on line 20
Type mismatch: bool expected but int found on line 20
Type mismatch: int expected but bool found on line 20
Type mismatch: int expected but bool found on line 20
Symbol 'a' undefined on line 21
Symbol 'b' undefined on line 21
Symbol 'foo' undefined on line 21
Symbol 'bar' undefined on line 21
Type mismatch: int expected but bool found on line 21
Symbol 'x' undefined on line 21
Symbol 'y' undefined on line 21
Type mismatch: int expected but bool found on line 21
Type mismatch: int expected but bool found on line 9
Symbol 'foo' undefined on line 22
Symbol 'a' undefined on line 22
Symbol 'bar' undefined on line 22
Symbol 'b' undefined on line 22
Type mismatch: int is incompatible with bool on line 22
Symbol 'arr' undefined on line 23
Symbol 'arr' undefined on line 23
Symbol 'arr' undefined on line 23
Symbol 'arr' undefined on line 23
Type mismatch: int expected but bool found on line 23
Symbol 'funcC' undefined on line 23
Symbol 'x' undefined on line 23
Symbol 'funcD' undefined on line 23
Symbol 'y' undefined on line 23
Type mismatch: bool expected but int found on line 24
Type mismatch: bool expected but int found on line 24
Type mismatch: int expected but bool found on line 24
Symbol 'foo' undefined on line 24
Symbol 'bar' undefined on line 24
Type mismatch: int is incompatible with bool on line 24
Type mismatch: int expected but bool found on line 23
Type mismatch: int expected but bool found on line 24
Symbol 'x' undefined on line 25
Symbol 'y' undefined on line 25
Symbol 'z' undefined on line 25
Symbol 'funcC' undefined on line 25
Symbol 'a' undefined on line 25
Symbol 'b' undefined on line 25
Symbol 'c' undefined on line 25
Symbol 'd' undefined on line 25
Type mismatch: bool expected but int found on line 25
Symbol 'foo' undefined on line 25
Symbol 'bar' undefined on line 25
Type mismatch: int expected but bool found on line 25
Type mismatch: bool expected but int found on line 23
Type mismatch: bool expected but int found on line 23
Type mismatch: int expected but bool found on line 22
Type mismatch: int expected but bool found on line 23
Type mismatch: void is incompatible with bool on line 4
